shared/analytics_tracking/__init__.py

```diff
@@ -1,171 +1,24 @@
-    or an organization.
-
-    For more context on how segment will store this data, see:
-    https://segment.com/docs/connections/warehouses/#identifies
-    """
-    if is_enterprise or not segment_enabled:
-        return
-
-    analytics.identify(user_id, user_data)
-
-
-def track_critical_files_sent(repoid, ownerid, commitid, pullid, is_enterprise):
-    track_event(
-        user_id=BLANK_SEGMENT_USER_ID,
-        event_name="Impact Analysis Critical Files Sent",
-        event_data={
-            "repo_id": repoid,
-            "repo_owner_id": ownerid,
-            "commit_id": commitid,
-            "pull_id": pullid,
-        },
-        is_enterprise=is_enterprise,
-    )
-
-
-def track_betaprofiling_added_in_YAML(repoid, ownerid, is_enterprise):
-    track_event(
-        user_id=BLANK_SEGMENT_USER_ID,
-        event_name="Impact Analysis Betaprofiling in YAML",
-        event_data={"repo_id": repoid, "repo_owner_id": ownerid},
-        is_enterprise=is_enterprise,
-    )
-
-
-def track_related_entrypoints_sent(repoid, ownerid, commitid, pullid, is_enterprise):
-    track_event(
-        user_id=BLANK_SEGMENT_USER_ID,
-        event_name="Impact Analysis Related Entrypoints Sent",
-        event_data={
-            "repo_id": repoid,
-            "repo_owner_id": ownerid,
-            "commit_id": commitid,
-            "pull_id": pullid,
-        },
-        is_enterprise=is_enterprise,
-    )
-
-
-def track_betaprofiling_removed_from_YAML(repoid, ownerid, is_enterprise):
-    track_event(
-        user_id=BLANK_SEGMENT_USER_ID,
-        event_name="Impact Analysis Betaprofiling removed from YAML",
-        event_data={"repo_id": repoid, "repo_owner_id": ownerid},
-        is_enterprise=is_enterprise,
-    )
-
-
-def track_show_critical_paths_added_in_YAML(repoid, ownerid, is_enterprise):
-    track_event(
-        user_id=BLANK_SEGMENT_USER_ID,
-        event_name="Impact Analysis Show Critical Files in YAML",
-        event_data={"repo_id": repoid, "repo_owner_id": ownerid},
-        is_enterprise=is_enterprise,
-    )
+__all__ = "analytics_tool"
 
 
-def track_show_critical_paths_removed_from_YAML(repoid, ownerid, is_enterprise):
-    track_event(
-        user_id=BLANK_SEGMENT_USER_ID,
-        event_name="Impact Analysis Show Critical Files removed from YAML",
-        event_data={"repo_id": repoid, "repo_owner_id": ownerid},
-        is_enterprise=is_enterprise,
-    )
+def get_list_of_analytic_tools() -> List[BaseAnalyticsTool]:
+    return [
+        Segment(),
+    ]
 
 
-def track_manual_critical_file_labelling_added_in_YAML(repoid, ownerid, is_enterprise):
-    track_event(
-        user_id=BLANK_SEGMENT_USER_ID,
-        event_name="Impact Analysis Manual Critical File Labelling in YAML",
-        event_data={"repo_id": repoid, "repo_owner_id": ownerid},
-        is_enterprise=is_enterprise,
-    )
+def get_analytics_tool() -> BaseAnalyticsTool:
+    for tool in get_list_of_analytic_tools():
```

Comment:
I think this should be configurable in the YAML. Similar to how the [CLI loads plugins](https://github.com/codecov/codecov-cli/blob/master/codecov_cli/plugins/__init__.py#L31)
 
That will allow users to use their own tracking tool without having to change our code much. (Although it can be argued that if you are hosting for yourself you don't _need_ a tracking tool, but whatever).

Then of course we will have our own implementations like you have there and use them as default